addons/main/functions/fnc_hasInjector.sqf

```diff
@@ -0,0 +1,10 @@
+#include "script_component.hpp"
+params ["_unit"];
+
+private _items = call FUNC(uniqueItems);
+
+private _availableInjectorItems = GVAR(injectorItems) arrayIntersect _items;
+if (_unit getUnitTrait "Medic" && {_availableInjectorItems isNotEqualTo []}) exitWith {
+    _unit setVariable [QGVAR(availableInjector), _availableInjectorItems select 0];
+    1
```

Comment:
if you only need to know if an injector is present then why not boolean, also you forgot the return value outside the exitwith